workflows/differentialabundance.nf

```diff
@@ -359,20 +383,31 @@ workflow DIFFERENTIALABUNDANCE {
         }
         .unique()
 
-    ch_all_matrices = VALIDATOR.out.sample_meta                 // meta, samples
-        .join(VALIDATOR.out.feature_meta)                       // meta, samples, features
-        .join(ch_raw)                                           // meta, samples, features, raw matrix
-        .combine(ch_processed_matrices)                         // meta, samples, features, raw, norm, ...
-        .map{
-            tuple(it[0], it[1], it[2], it[3..it.size()-1])
-        }
-        .first()
-
-    ch_contrast_variables
-        .combine(ch_all_matrices.map{ it.tail() })
+    if(params.study_type != "geo_soft_file") {
```

Comment:
You don't need to duplicate this whole section. You can just do like:

```
if(params.study_type != "geo_soft_file") {
   ch_mat = ch_raw.combine(ch_processed_matrices)
}else{
   ch_mat = ch_processed_matrices
}
```

... and then use `ch_mat` in the original chain of ops

Reply:
Makes a lot more sense... thanks and sorry for the bad code!